pkgs/development/libraries/oracle-instantclient/default.nix

```diff
@@ -1,71 +1,130 @@
-  }."${stdenv.hostPlatform.system}" or throwSystem;
+  # calculate the filename of a single zip file
+  srcFilename = component: arch: version: rel:
+    "instantclient-${component}-${arch}-${version}" +
+    (optionalString (rel != "") "-${rel}") +
+    (optionalString (arch == "linux.x64") "dbru") + # ¯\_(ツ)_/¯
+    ".zip";
+
+  # fetcher for the clickthrough artifacts (requiring manual download)
+  fetchClickThrough =  srcFilename: hash: (requireFile {
+    name = srcFilename;
+    url = "https://www.oracle.com/database/technologies/instant-client/downloads.html";
+  });
+
+  # fetcher for the non clickthrough artifacts
+  fetchSimple = srcFilename: hash: fetchurl {
+    url = "https://download.oracle.com/otn_software/linux/instantclient/193000/${srcFilename}";
+    sha256 = hash;
+  };
+
+  # pick the appropriate fetcher depending on the platform
+  fetcher = if stdenv.hostPlatform.system == "x86_64-linux" then fetchSimple else fetchClickThrough;
 
+  # assemble srcs
+  srcs = map (component:
+    (fetcher (srcFilename component arch version "") hashes."${component}" or ""))
+  components;
+
+  pname = "oracle-instantclient";
   extLib = stdenv.hostPlatform.extensions.sharedLibrary;
-in stdenv.mkDerivation rec {
-  inherit version srcs;
-  name = "oracle-instantclient-${version}";
+in stdenv.mkDerivation {
+  inherit pname version srcs;
 
   buildInputs = [ stdenv.cc.cc.lib ]
-    ++ optionals (stdenv.isLinux) [ libaio ]
+    ++ optional stdenv.isLinux libaio
     ++ optional odbcSupport unixODBC;
 
   nativeBuildInputs = [ makeWrapper unzip ]
     ++ optional stdenv.isLinux autoPatchelfHook
     ++ optional stdenv.isDarwin fixDarwinDylibNames;
 
+  outputs = [ "out" "dev" "lib"];
+
   unpackCmd = "unzip $curSrc";
 
   installPhase = ''
-    mkdir -p "$out/"{bin,include,lib,"share/java","share/${name}/demo/"}
-    install -Dm755 {sqlplus,adrci,genezi} $out/bin
-    ${optionalString stdenv.isDarwin ''
-      for exe in "$out/bin/"* ; do
-        install_name_tool -add_rpath "$out/lib" "$exe"
-      done
-    ''}
-    ln -sfn $out/bin/sqlplus $out/bin/sqlplus64
-    install -Dm644 *${extLib}* $out/lib
+    mkdir -p "$out/"{bin,include,lib,"share/java","share/${pname}-${version}/demo/"} $lib/lib
+    install -Dm755 {adrci,genezi,uidrvci,sqlplus} $out/bin
+    install -Dm644 *${extLib}* $lib/lib
     install -Dm644 *.jar $out/share/java
     install -Dm644 sdk/include/* $out/include
-    install -Dm644 sdk/demo/* $out/share/${name}/demo
+    install -Dm644 sdk/demo/* $out/share/${pname}-${version}/demo
+
+    # provide alias
+    ln -sfn $out/bin/sqlplus $out/bin/sqlplus64
 
     # PECL::oci8 will not build without this
     # this symlink only exists in dist zipfiles for some platforms
-    ln -sfn $out/lib/libclntsh${extLib}.12.1 $out/lib/libclntsh${extLib}
+    ln -sfn $lib/lib/libclntsh${extLib}.12.1 $lib/lib/libclntsh${extLib}
```

Comment:
`12.1` would need to be updated here I imagine.

Reply:
Not sure. The package ships multiple versions:

```
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntsh.so
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntsh.so.10.1
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntsh.so.11.1
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntsh.so.12.1
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntsh.so.18.1
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntsh.so.19.1
/nix/store/a9bwgf3w7xq1019kacjbwkfssp1g7n7y-oracle-instantclient-19.3.0.0.0-lib/lib/libclntshcore.so.19.1
```

I think the better fix would be to get PECL::oci8 to `dlopen`/link against the version it works with.

Reply:
building on current master and `readelf -d` at the `oci8.so` shows it only needs the followin libraries:

```
 0x0000000000000001 (NEEDED)             Shared library: [libclntsh.so.12.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
```

so we can drop the `libclntsh.so` symlink.

However, there's another problem in `phpPackages.oci8` - it didn't handle oracle-instantclient being a split derivation. Will upload a patch fixing this.